core/kernel/src/feature_manifest.rs

```diff
@@ -0,0 +1,270 @@
+use serde::Deserialize;
+use std::collections::HashMap;
+
+#[derive(Deserialize, PartialEq, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)] // Temporary until exposed through C API
+pub struct FeatureManifest {
+    name: Option<String>,
+    author: Option<String>,
+    version: Option<String>,
+    description: Option<String>,
+    thumbnail: Option<String>,
+    thumbnail_color: Option<String>,
+    primary_color: Option<String>,
+    links: Option<HashMap<String, String>>,
+}
+
+#[derive(PartialEq, Debug)]
+pub struct FeatureManifestParsingError {
+    description: String,
+}
+
+// Alias for str, probably will be moved to a centralized place to be reused
+pub type JSONStr = str;
+
+#[allow(dead_code)] // Temporary until exposed through C API
+impl FeatureManifest {
+    pub fn parse(
+        json: &JSONStr,
+        locale: &str,
+    ) -> Result<FeatureManifest, FeatureManifestParsingError> {
+        FullFeatureManifest::try_from(json)
+            .map(|manifest| FeatureManifest::build_feature_manifest(manifest, locale))
+    }
+
+    fn build_feature_manifest(full_manifest: FullFeatureManifest, locale: &str) -> FeatureManifest {
+        let translation = full_manifest
+            .translations
+            .as_ref()
+            .and_then(|unwrapped| unwrapped.get(locale));
+
+        match translation {
+            Some(translation) => {
+                let mut links = full_manifest.links.unwrap_or_default();
+                if let Some(translated_links) = translation.links.clone() {
+                    links.extend(translated_links.into_iter());
+                }
+                FeatureManifest {
+                    name: translation
+                        .name
+                        .clone()
+                        .or(full_manifest.name),
+                    author: translation
```

Comment:
not clear what you're doing here. You're using the translation on *everything*? I would say only description will need it...

Reply:
I am following the spec from iOS/Android, where all fields can be translated. And it makes sense, the only fields that may not be translated are `color` field.

Reply:
`author` too? `name` OK, there might be be several versions of the name... But I don't see examples of translations in the rest. Again, the issue is the lack of spec. Reverse engineering from code is always a complicated thing.

Reply:
Well yes, the name of the author would be different in Russian, for example. Again I am speculating here, there is no clear contract describing what should go into translations.

Also, to me it makes sense to have the `thumbnail` translation, especially if it can contain text.

Reply:
It's simply an icon. Again, I think we should agree on some kind of spec first.

Reply:
I see we have added the "Beta" mark in the thumbnail which contains text.

Then let's get this going and have a JSON schema created. I can sketch one, and then we can iterate over it and polish to whatever state we consider acceptable.

Reply:
I have added the JSON schema spec here https://wiki.polypoly.eu/display/POLYPOD/Feature+manifest+spec